frontend/src/lib/codemirror/languageServer.ts

```diff
@@ -411,6 +419,18 @@ class LanguageServerPlugin implements PluginValue {
         }
     }
 
+    async requestFormat() {
+        return await this.client.textDocumentFormatting({
+            textDocument: {
+                uri: this.documentUri,
+            },
+            options: {
```

Comment:
I presume clangd uses clang-format - do we know how to provide full clang-format options? Do we need to create a `.clang-format` vfs file or something?

Reply:
Yup; that's exactly it. I've added support to clangd-wasm to create `compile-commands.json` somewhat elegantly; it probably makes sense to treat `.clang-format` the same way.

to pick your brain: I'm going back on forth on the best way to expose the format settings. A "decomp.me profile" that's baked into the site? Preset-specific settings? Just exposing the whole thing to end users? (get an LLM to guess based on context and the weather? :))

Reply:
I think exposing the whole thing to end users is fine, many projects (presets) will have an existing clang-format config that they would do well to just copy paste in. We can probably format config this for a later enhancement and just use a sensible default for now (make an issue?) unless you want to tackle it

Reply:
I think a ClangFormat model that just holds the yaml as a string would be fine, and we can make CompilerConfig (#614) optionally reference one. If there is no ClangFormat referenced we can use a decomp.me default or a user setting default

Reply:
As a methodology; that seems pretty sound!
There's enough left on the todo list that unless someone wants to vouch for this as critical, I'll probably move it onto the future brush-up pile. I'll wire up a default `.clang-format` for this PR, which should tee things up nicely for future work.

Reply:
pmret/papermario has a .clang-format that we like that you could use

Reply:
oh nice, you already did lol